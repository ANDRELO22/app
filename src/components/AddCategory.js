
import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCat}) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleInputChange =(e)=>{
        setInputValue(e.target.value);
    }

    const hendleSumit =(e) => {
        //esto no deja que cada vez que le des entre se refresque toda la paguina 
        e.preventDefault();

        //esto no deja que lo que se escriba en la caja de texto sea menor a dos letras
        if( inputValue.trim().length > 2 ){
             //console.log('Sumit echo');
             //asi es como se cominica con el otro componente
        setCat(cats =>[inputValue,...cats ]);
        setInputValue('');
        }

        
    }
  return (
        <form onSubmit={hendleSumit}>
        <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}

        /> 

        </form>
  )
};

AddCategory.propTypes={
    setCat: PropTypes.func.isRequired
}