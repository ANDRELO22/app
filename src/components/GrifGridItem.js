import PropTypes from 'prop-types'
import React from 'react';

export const GrifGridItem = ({ title, url}) => {

    //console.log({title,url});
  return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
  )
};
 GrifGridItem.propTypes={

  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired

 }

