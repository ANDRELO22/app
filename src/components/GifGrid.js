
import React, {/*useState, useEffect*/} from 'react';
import {useFectchGifs} from '../hooks/useFectchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GrifGridItem } from './GrifGridItem';

export const GifGrid = ({category}) => {

  //const [caunt, setCaunt] = useState(0);

  //const [images, setImages] = useState([ ]);
  //esto le dice al codigo que serenderise solo por primera vez y no cada ves que no note un cambio
    //useEffect( () => {
    //getGifs( category )
    //.then(setImages)
  //},[ category ])

  //un ejemplo
  const {data:images, loading} = useFectchGifs(category);

    
    


  return (
        <>
        <h3 className="animate__animated animate__bounce"> { category }</h3>

        {loading && <p className="animate__animated animate__flash">Esperame bro voy en camino...</p> }
     
        
        <div className="card-grid ">
        
        
        
        {
        images.map( img =>(
          <GrifGridItem
          key= {img.id}
          {...img}
          />
        ))
        }
        
        
  </div>
      
      </>
  )
};
