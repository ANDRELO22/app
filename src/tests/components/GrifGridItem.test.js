import React from 'react'
import  { shallow } from 'enzyme';
import { GrifGridItem } from '../../components/GrifGridItem';


describe('pruebas en <GrifGridItem />', () =>{

    const title= 'un titulo¡¡';
    const url = 'http://localhost/cualquiercosa.jpg'
    const wrapper = shallow(<GrifGridItem  title={title} url={url} />);
    test('debe mostrar el componente correctamente', () => {
        
    
    expect(wrapper).toMatchSnapshot();
    })
    test('debe tener un parrafo con el titulo', () => { 
        const p = wrapper.find('p')
        expect( p.text().trim()).toBe( title )
    })

    test('debe de tener la imagen igal al url y el alt de las prop', () => { 
        const img = wrapper.find('img')
        console.log(img.props('src'))
    })
}) 