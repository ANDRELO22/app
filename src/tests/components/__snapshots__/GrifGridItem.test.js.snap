// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GrifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="un titulo¡¡"
    src="http://localhost/cualquiercosa.jpg"
  />
  <p>
    un titulo¡¡
  </p>
</div>
`;
